projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/index.js

```diff
@@ -32,9 +32,11 @@ export const settings = {
 		const [ dontShowAgain, setDontShowAgain ] = useState( false );
 		const [ isChecked, setIsChecked ] = useState( false );
 
-		const { launchpadScreenOption, siteIntentOption } = window?.Jetpack_LaunchpadSaveModal || {};
+		const { launchpadScreenOption, hasNeverPublishedPostOption, siteIntentOption } =
+			window?.Jetpack_LaunchpadSaveModal || {};
 		const isInsideSiteEditor = document.getElementById( 'site-editor' ) !== null;
 		const isInsidePostEditor = document.querySelector( '.block-editor' ) !== null;
+		const prevHasNeverPublishedPostOption = useRef( hasNeverPublishedPostOption );
```

Comment:
Out of curiosity, why are we using useRef here to track this, vs just state?

Reply:
No strong reasoning. I was following previous conventions established in the file  https://github.com/Automattic/jetpack/pull/28989/files#diff-13206839dd3315d528c316c3405fbf89039de38e648bc7b16ac786db91fbb37dR25-R26

Happy to switch to state management instead

Reply:
Yes, I'm trying to understand if it matters one way or the other. I usually think of the key difference being that changes to state values cause the component re-render while changes to ref values do not. Or to put it differently, does the behavior of the component change when the underlying value changes? 

Does all that matter one way or the other here?